app/controllers/application_controller.rb

```diff
@@ -5,7 +5,7 @@ class ApplicationController < ActionController::Base
 
   def ensure_authentication_allowed
     if Feature.disabled?(:password_authentication) && Feature.disabled?(:google_authentication)
-      redirect_to root_path, alert: "Password and Google authentication are both disabled."
+      flash[:alert] = "Password and Google authentication are both disabled."
```

Comment:
Wouldn't it be better to render a classical 401 page instead?

E.g. an exception app or `rescue_from` ...

Reply:
Yes, good point. That's much simpler.